### PR TITLE
Add test_build.py based on test_iss.py

### PR DESCRIPTION
**Pull request by joshmoore**
Depends on gh-515

Intended to be equivalent to [test2.sh](https://github.com/joshmoore/build-starfish/commit/0cee6d92aebf5f90a2fc3eff5394e370438b99c7)

### CODE REVIEW

**Comment by joshmoore** [comment]
Using a merged build (from [86c4a2a83d4](https://github.com/joshmoore/build-starfish/commit/86c4a2a83d49ec1e99b40e52c9ad1c39a8a38bbe)):

```
[jamoore@idr1-slot2 build-starfish]$ docker-compose run all -svxk test_build
...
starfish build  ==> 1.5559282302856445 seconds
/home/starfish/.conda/envs/env/lib/python3.6/site-packages/validate_sptx/util.py:78: UserWarning:
 'codebook' is a required property
	Schema:         	https://github.com/spacetx/sptx-format/schema/experiment.json
	Subschema level:	0
	Path to error:  	required

  warnings.warn(message)
/home/starfish/.conda/envs/env/lib/python3.6/site-packages/validate_sptx/util.py:78: UserWarning:
 {} is not of type 'array'
	Schema:         	https://github.com/spacetx/sptx-format/schema/codebook/codebook.json
	Subschema level:	0
	Path to error:  	type

  warnings.warn(message)
FAILED
```

i.e. how do we get a codebook into this test?

**Comment by joshmoore** [comment]
Does anyone have a preference? My guess would be that generating the minimal codebook allowed by the schema would be sufficient. Perhaps with a target string of "PLEASE_REPLACE_ME"

File: starfish/test/full_pipelines/cli/test_build.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    def test_run_build(self):
+        tempdir = tempfile.mkdtemp()
+        coverage_enabled = "STARFISH_COVERAGE" in os.environ
+
+        def callback(interval):
+            print(" ".join(stage[:2]), " ==> {} seconds".format(interval))
+
+        try:
+            # TODO: duplicated, time to refactor
```
Comment (ttung) [inline starfish/test/full_pipelines/cli/test_build.py]: +1, file issue?

File: starfish/test/full_pipelines/cli/test_build.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            lambda tempdir: tempdir
+        ],
+        [
+            "validate-sptx", "--experiment-json",
+            lambda tempdir: os.sep.join([tempdir, "experiment.json"])
+        ],
+    )
+
+    def test_run_build(self):
+        tempdir = tempfile.mkdtemp()
```
Comment (ttung) [inline starfish/test/full_pipelines/cli/test_build.py]: Can you replace this with `tempfile.TemporaryDirectory`?  The code you're copying from preceded the decision to support only python 3.

Comment (joshmoore) [reply starfish/test/full_pipelines/cli/test_build.py]: Line removed in favor of `exec.stages`.

File: starfish/test/full_pipelines/cli/test_build.py (hunk 1 above)
Comment (joshmoore) [reply starfish/test/full_pipelines/cli/test_build.py]: No longer duplicated.

**Comment by joshmoore** [comment]
@ttung : re-review request

**Pull request merged**